README.md

```diff
@@ -1193,8 +1194,9 @@ class Color
   end
 
   def self.parse(value)
-    fail 'Invalid color' unless %w(blue red green).include?(value)
-    new(value)
+    return new(value) if %w[blue red green]).include?(value)
+
+    Grape::Types::InvalidValue.new('Unsupported color')
```

Comment:
I feel like this should be a `raise` that gets handled, not a return.

Reply:
How will we know that an exception message should be returned to the user if we rescue `StandardError`? I would like to avoid guessing :slightly_smiling_face:  We already return the invalid value [there](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/types/custom_type_coercer.rb#L58) and [there](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/types/dry_type_coercer.rb#L67) and [there](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/types/array_coercer.rb#L41).

In general, if there is an easy way to return a value indicating a failure, I prefer to do that. It is cheaper:

```ruby
require 'benchmark/ips'

def return_value
  :invalid
end

def raise_exception
  raise StandardError
end

Benchmark.ips do |x|
  x.report('return value') do
    return_value
  end
  x.report('raise error') do
    begin
      raise_exception
    rescue StandardError
    end
  end

  x.compare!
end
```

```
Warming up --------------------------------------
        return value     1.990M i/100ms
         raise error   149.703k i/100ms
Calculating -------------------------------------
        return value     19.948M (± 0.4%) i/s -    101.470M in   5.086841s
         raise error      1.513M (± 0.5%) i/s -      7.635M in   5.044907s

Comparison:
        return value: 19947800.7 i/s
         raise error:  1513417.9 i/s - 13.18x  (± 0.00) slower
```

Reply:
Ok, I clearly didn't do my homework, especially on the existing code.